File: frontend/projects/api/src/lib/models/today-dto.ts
import { CheckInDto } from './check-in-dto';
import { TodayFastingDto } from './today-fasting-dto';
import { TodayHealthDto } from './today-health-dto';
import { TodayPeopleDto } from './today-people-dto';
import { TodayReadingDto } from './today-reading-dto';
import { TodayStreaksDto } from './today-streaks-dto';
import { VerseDto } from './verse-dto';

/**
 * The aggregate behind `GET /api/today` — one round trip for the whole
 * dashboard. Each pillar nests its own object so later pillars (people,
 * gratitude) arrive as new nullable properties without breaking this shape.
 */
export interface TodayDto {
  readonly date: string;
  readonly greetingName: string;
  readonly checkIn: CheckInDto | null;
  readonly verse: VerseDto;
  readonly reading: TodayReadingDto | null;
  readonly streaks: TodayStreaksDto;
  readonly fasting: TodayFastingDto;
  readonly health: TodayHealthDto;
  readonly people: TodayPeopleDto;
}

File: backend/src/Shalom.Application/Contracts/TodayDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// The aggregate powering the Today dashboard in one round trip (and the
/// single offline-cached resource).
///
/// Versioning rule: each pillar nests its own object (checkIn, verse,
/// reading, streaks) so later pillars — fasting, workout, people — arrive as
/// new nullable properties without changing any existing shape.
/// </summary>
public record TodayDto(
    DateOnly Date,
    string GreetingName,
    CheckInDto? CheckIn,
    VerseDto Verse,
    TodayReadingDto? Reading,
    TodayStreaksDto Streaks,
    TodayFastingDto Fasting,
    TodayHealthDto Health,
    TodayPeopleDto People,
    bool RitualCompletedToday);
